geo-types/src/geometry.rs

```diff
@@ -182,94 +181,56 @@ impl<T: CoordNum> Geometry<T> {
     }
 }
 
-#[derive(Debug)]
-pub struct FailedToConvertError;
-
-impl fmt::Display for FailedToConvertError {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "Could not convert from enum member to concrete type")
-    }
-}
-
-impl Error for FailedToConvertError {
-    fn description(&self) -> &str {
-        "Could not convert from enum member to concrete type"
-    }
-}
-
-impl<T: CoordNum> TryFrom<Geometry<T>> for Point<T> {
-    type Error = FailedToConvertError;
-
-    fn try_from(geom: Geometry<T>) -> Result<Point<T>, Self::Error> {
-        match geom {
-            Geometry::Point(p) => Ok(p),
-            _ => Err(FailedToConvertError),
+#[macro_use]
+macro_rules! try_from_geometry_impl {
+    ($($type: ident),+) => {
+        $(
+        /// Convert a Geometry enum into its inner type.
+        ///
+        /// Fails if the enum case does not match the type you are trying to convert it to.
+        impl <T: CoordNum> TryFrom<Geometry<T>> for $type<T> {
+            type Error = Error;
+
+            fn try_from(geom: Geometry<T>) -> Result<Self, Self::Error> {
+                match geom {
+                    Geometry::$type(g) => Ok(g),
+                    other => Err(Error::MismatchedGeometry {
+                        expected: type_name::<$type<T>>(),
+                        found: inner_type_name(other)
+                    })
+                }
+            }
         }
+        )+
     }
 }
 
-impl<T: CoordNum> TryFrom<Geometry<T>> for Line<T> {
-    type Error = FailedToConvertError;
+try_from_geometry_impl!(
```

Comment:
I'm starting to leverage more macros to DRY up our code. I'm pretty new to using macros, so if this feels sophomoric or something, let me know - I'm not actually that bothered by the mostly copy-pasted implementations that I replaced.

Reply:
I am completely clueless when it comes to writing macros but this one is readable and I understand what it's doing, so I think that's a good sign?